custom/enikshay/reports/views.py

```diff
@@ -39,3 +46,31 @@ def get(self, request, domain, *args, **kwargs):
                 'total': location_choice_provider.query_count(query_context.query, user)
             }
         )
+
+
+@domain_admin_required
+def duplicate_ids_report(request, domain, case_type):
+    case_type = {'voucher': CASE_TYPE_VOUCHER, 'person': CASE_TYPE_PERSON}[case_type]
+    id_property = {'voucher': VOUCHER_ID, 'person': ENIKSHAY_ID}[case_type]
+
+    accessor = CaseAccessors(domain)
+    case_ids = accessor.get_case_ids_in_domain(case_type)
+    all_cases = [
+        {
+            'case_id': case.case_id,
+            'readable_id': case.get_case_property(id_property),
+            'opened_on': case.opened_on,
+        }
+        for case in accessor.iter_cases(case_ids)
+    ]
```

Comment:
Would this eventually become a very long list in case of many records. 
Storing all cases and then iterating on them twice could hurt. 

You probably can store the corresponding case_id(s) for a readable_id and just load those cases to find the bad cases, this would save one/two iterations over all cases. May be something like:
```
bad_case_ids = []
id_property_with_case_ids = {}
id_property_with_case_ids = defaultdict{lambda:[], id_property_with_case_ids}
for case in accessor.iter_cases(case_ids):
    id_prop_value = case.get_case_property(id_property)
    if id_prop_value:  # to avoid empty values
        id_property_with_case_ids[case.get_case_property(id_property)].append(case.case_id)
for id_prop_value in id_property_with_case_ids:  # iterate over just non-empty values and not all the cases
    if len(id_property_with_case_ids[id_prop_value]) > 2:
        bad_case_ids.append(id_property_case_ids[id_prop_value])

bad_cases = accessor.get_cases(bad_case_ids)
```

Reply:
It transforms each case into a 3 element dict and then takes two more passes over that, so there's never a list of the full cases in memory.  Think that's fine?

Reply:
Also, for scale, this should be on the order of 100k elements total, and I think a few thousand with duplicates.

Reply:
`all_cases` has dict for all cases in domain right? I didnt mean all cases in memory but just a big object anyway.
Memory was one point but also the number of iterations.

Plus if there is a way to reduce the number of iterations to speed this up, we should do it anyway?
Also, Do we need to skip blank values?

Reply:
I'm not convinced.  With python primitives, this should be quite small.  I just did some back-of-the envelope testing to validate:
```python
In [1]: import timeit

In [2]: a = [{'foo': 'bar', 'baz': 'bang', 'bip': 'fuzz'} for i in range(100000)]

In [3]: a.__sizeof__()
Out[3]: 824440

In [4]: def walk():
   ...:     for x in a:
   ...:         pass
   ...:     

In [5]: timeit.timeit(walk, number=100)
Out[5]: 0.32859110832214355
```
That is, this list of 100k dicts of 3 elements each has a size less than a MB.  Iterating over that list 100 times takes about a third of a second.  For a debugging report that will be used only infrequently, I think that is acceptable.  For something like this, I'd heavily weight code clarity over marginal perfomance gains.

Reply:
Regarding blank values - I think it makes sense to keep those in, as that's also an error AFAIK, so it couldn't hurt to have some visibility into those.

Reply:
Thank you Ethan. 🙇